### PR TITLE
Firefox fixes

### PR DESCRIPTION
**Pull request by bemasc**
This change
- allows sending Blobs on Firefox, which can't use FileReader (until FF46, due in April)
- adds integration test coverage for Firefox

This change depends on changes to Freedom (and Freedom-for-Firefox and Chrome) that have not yet been released.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -8,9 +8,7 @@ module.exports = function (grunt) {
   require('time-grunt')(grunt);
 
   // load all grunt tasks
-  require('load-grunt-tasks')(grunt, {
-    pattern: ['grunt-*', '!grunt-template-jasmine-istanbul']
-  });
+  require('jit-grunt')(grunt, {});
```
Comment (agallant) [inline Gruntfile.js:11]: `jit-grunt` is great, but a potential gotcha - if the npm module is named differently from the grunt task then you have to give it the mapping. I actually think that's not a problem for your current tasks, but if you add e.g. `grunt-npm` then you'd need a line like this: https://github.com/freedomjs/freedom-pgp-e2e/blob/master/Gruntfile.js#L15

File: src/xhr.corexhr.js [hunk 2]
```diff
@@ -237,11 +237,7 @@ XhrShim.prototype.send = function(data) {
   } else if (data instanceof ArrayBuffer) {
     this.xhr_.send({buffer: data});
   } else if (data instanceof Blob) {
-    var reader = new FileReader();
-    reader.onload = function() {
-      this.xhr_.send({buffer: reader.result});
-    }.bind(this);
-    reader.readAsArrayBuffer(data);
+    this.xhr_.send({blob: data});
```
Comment (agallant) [inline src/xhr.corexhr.js:240]: Would there be any purpose to still using FileReader here in case it's _not_ Firefox? Or if Blob is good enough, why not just stick with Blob even when FileReader is supported by Firefox?

Comment (bemasc) [inline src/xhr.corexhr.js:240]: AFAIK, Blob is fine, so I don't expect to undo this change.  The integration tests prove that this works on Chrome too, and Freedom uses blobs in other core APIs (e.g. core.websocket) already.

**Comment by agallant** [comment]
:+1:
